src/test/resources/testPrograms/patternMatching/CaseClasses.monty

```diff
@@ -0,0 +1,62 @@
+println("--- haskell-style lists ---")
+
+abstract class List:
+    + Int getLength():
+        case self of:
+            Cons(Int value, List l):
```

Comment:
Sadly no.
Currently, I don't know a solution for that problem, since in this case the base class ``List`` is not aware of all of its children. It would be possible to define a further case class (besides ``Cons`` and ``Empty``) that inherits from ``List``. The compiler would then also have to consider this case. 
It becomes even more weird if one would inherit a case class from another case class.
Also, always requiring a wildcard would restrict the programmer (since the wildcard is not able to bind any name).

As a last point, the ``ControlFlowVisitor`` has a bug regarding ``if``-``else`` statements in functions. Consider the following function:

    Bool even(Int x):
        if x%2 = 0:
            return true
        else:
            return false

The compiler would complain that there is a return statement missing, while the following function works as expected:

    Bool even(Int x):
        if x%2 = 0:
            return true
        return false

I tried to fix that bug, but found that this was not trivial (for some reason I don't really remember).
Since I didn't introduce this but, I didn't really feel responsible to fix it (maybe I'll do that later, when there are less important things to do). However, this has an effect on the implementation of pattern matching, since it uses ``if``-``else`` statements.